### PR TITLE
[PropertyInfo] Give precedence to constructor arguments over mutators

### PR DESCRIPTION
**Pull request by SCIF**
| Q             | A
| ------------- | ---
| Branch?       | 4.3
| Bug fix?      | yes
| New feature?  |no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #33415
| License       | MIT
| Doc PR        | --

### CODE REVIEW

File: src/Symfony/Component/PropertyInfo/Extractor/ReflectionExtractor.php [hunk 1]
```diff
@@ -131,8 +131,14 @@ public function getProperties($class, array $context = []): ?array
     /**
      * {@inheritdoc}
      */
-    public function getTypes($class, $property, array $context = []): ?array
+    public function getTypes($class, $property, array $context = [])
```
Comment (linaori) [inline src/Symfony/Component/PropertyInfo/Extractor/ReflectionExtractor.php]: Shouldn't this return type stay?

Comment (SCIF) [reply src/Symfony/Component/PropertyInfo/Extractor/ReflectionExtractor.php]: My bad. Just copy and pasted over different branches.

File: src/Symfony/Component/PropertyInfo/Tests/Fixtures/Php71Dummy.php [hunk 2]
```diff
@@ -51,3 +51,14 @@ public function getIntWithAccessor()
     {
     }
 }
+
+class ImmutableTest {
```
Comment (nicolas-grekas) [inline src/Symfony/Component/PropertyInfo/Tests/Fixtures/Php71Dummy.php]: bracket on the next line please

File: src/Symfony/Component/PropertyInfo/Tests/Fixtures/Php71Dummy.php [hunk 3]
```diff
@@ -51,3 +51,14 @@ public function getIntWithAccessor()
...
+    public function __construct(?int $test = null)
+    {
+    }
+
+    public function getTest(): int
+    {
+
```
Comment (nicolas-grekas) [inline src/Symfony/Component/PropertyInfo/Tests/Fixtures/Php71Dummy.php]: empty line should be removed

File: src/Symfony/Component/PropertyInfo/Tests/Fixtures/Php71Dummy.php [hunk 4]
```diff
@@ -51,3 +51,14 @@ public function getIntWithAccessor()
     {
     }
 }
+
+class ImmutableTest {
+    public function __construct(?int $test = null)
```
Comment (nicolas-grekas) [inline src/Symfony/Component/PropertyInfo/Tests/Fixtures/Php71Dummy.php]: please remove the `?`

**Review by nicolas-grekas — COMMENTED** [review]
if it's a bug fix, it's for 4.3
if it's a BC break, 4.4 won't fit, neither master
we need to balance this /cc @dunglas

**Comment by SCIF** [comment]
Hi @nicolas-grekas , I've changed base for this PR and fixed review notices.

**Comment by fabpot** [comment]
Closing as this would be a BC break. I've suggested one workaround (to be verified) in the linked issue that I won't close for now. Let's continue the discussion there.

**Pull request closed without merging**